execution/aggregate/scalar_table.go

```diff
@@ -120,17 +138,29 @@ func makeAccumulatorFunc(expr parser.ItemType) (newAccumulatorFunc, error) {
 	case "sum":
 		return func() *accumulator {
 			var value float64
-			var hasValue bool
+			var histSum *histogram.FloatHistogram
+			var hasFloatVal bool
 
 			return &accumulator{
-				AddFunc: func(v float64) {
-					hasValue = true
-					value += v
+				AddFunc: func(v float64, h *histogram.FloatHistogram) {
+					if h == nil {
+						hasFloatVal = true
+						value += v
+						return
+					}
+					if histSum == nil {
+						histSum = h
+						return
+					}
+					histSum = histSum.Add(h)
+				},
+				ValueFunc: func() (float64, *histogram.FloatHistogram) {
+					return value, histSum
 				},
-				ValueFunc: func() float64 { return value },
-				HasValue:  func() bool { return hasValue },
+				// Sum returns an empty result when floats are histograms are aggregated.
```

Comment:
Do you mean this here?:
```suggestion
				// Sum returns an empty result when floats and histograms are aggregated.
```